core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -352,17 +352,36 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     val ui = SparkUI.create(None, new HistoryAppStatusStore(conf, kvstore), conf, secManager,
       app.info.name, HistoryServer.getAttemptURI(appId, attempt.info.attemptId),
       attempt.info.startTime.getTime(), attempt.info.appSparkVersion)
-    loadPlugins().foreach(_.setupUI(ui))
-
+    setupPluginUI(ui)
     val loadedUI = LoadedAppUI(ui)
-
     synchronized {
       activeUIs((appId, attemptId)) = loadedUI
     }
 
     Some(loadedUI)
   }
 
+  private def setupPluginUI(ui: SparkUI): Unit = {
+    val plugins = loadPlugins().toSeq
+
+    var sqlTab: Option[AppHistoryServerPlugin] = None
+    var jdbcTab: Option[AppHistoryServerPlugin] = None
+    val otherTabs: Seq[AppHistoryServerPlugin] = Seq()
+
+    plugins.foreach{ plugin =>
+      if (plugin.toString.contains("SQLHistoryServerPlugin")) {
```

Comment:
Thanks @gengliangwang ,By default, JDBC/ODBC tab comes first then SQL tab which is different from live UI. To make sure SQL tab comes first I created the method.

Reply:
I think this is kind of ugly. How about just rename the `HiveThriftServer2HistoryServerPlugin`, e.g. `ThriftServer2HistoryServerPlugin`, so that we don't need to do such hard code matching.

Reply:
Thanks, let me try that

Reply:
Seems the class name change won't help here. I am not sure any better way than this. 
Earlier I did `loadPlugins().toSeq.reverse.foreach(_.setupUI(ui))`, which also worked.

![Screenshot 2019-11-27 at 6 18 07 AM](https://user-images.githubusercontent.com/23054875/69684249-f73cdf80-10dd-11ea-8c7b-1c7cf14c1df9.png)

Reply:
Yes, that's is kind of expected result...
Then maybe we should leave it as it is.
Please add comments as well.

Reply:
Done.

Reply:
BTW what did you changed `loadPlugins().toSeq.reverse.foreach(_.setupUI(ui))`. I think it is better, and hard-coding here can get challenged by other reviewers.

Reply:
I think I changed based on the comment, https://github.com/apache/spark/pull/26378#discussion_r349675820 . Let me check if any better way is there.

Reply:
@gengliangwang Another way could be, control the tab position based on class name
```
    plugins.map { plugin =>
      val splitPath: Array[String] = plugin.toString.split("\\.")
      val className = splitPath(splitPath.length - 1)
      (plugin, className)
    }.sortBy(_._2).foreach(_._1.setupUI(ui))
```

Here, we just need to make sure, whatever tab need to come later, has greater className than the previous one.
Here if I make  class name as `ThriftServer2HistoryServerPlugin`, then it will come at the end. But I am not sure, we need to make it complicated, compare to the exisitng one, as long as we comment it properly.

Reply:
I prefer not to hard code matching the "sql"/"thriftserver" plugin here. Let's sort the plugin by its simple name (`plugin.getClass.getSimpleName`).